### PR TITLE
feat: add --inclusive-tpot support to estimate and recommend modes

### PR DESCRIPTION
**Pull request by natoscott**
## Overview

Extends the `--inclusive-tpot` flag (originally added for default and exp modes in #1141) to estimate and recommend modes, enabling benchmark-comparable TPOT values across all CLI modes and Python API functions.

### What changed

1. **Core utilities** (`src/aiconfigurator/cli/report_and_save.py`):
   - Added `get_inclusive_tpot()` for scalar transformations
   - Complements existing `_apply_inclusive_tpot()` for DataFrames

2. **Estimate mode** (`src/aiconfigurator/cli/main.py`):
   - Added `--inclusive-tpot` CLI argument
   - Transforms display TPOT before printing (presentation layer only)
   - Handles all estimate submodes: agg, disagg, afd, static, static_ctx, static_gen

3. **Recommend mode** (`src/aiconfigurator/cli/api.py`, `main.py`):
   - Added `--inclusive-tpot` CLI argument
   - Threaded flag through: `_run_recommend()` → `cli_recommend()` → `_execute_and_wrap_result()`
   - Also added to `cli_default()` and `cli_exp()` for Python API users
   - Updated all mock_args for `save_results()` compatibility

4. **Tests** (`tests/unit/cli/`):
   - 10 unit tests for `get_inclusive_tpot()` scalar utility
   - 4 argument parsing tests (estimate + recommend modes)
   - All tests passing

5. **Documentation** (`docs/cli_user_guide.md`):
   - Updated Inclusive TPOT section with estimate and recommend examples
   - Added Python API usage examples to docstrings

### Design principles

- **Output transformation only** - never modifies internal calculations
- **TTFT always unchanged**
- Formula: `(ttft + tpot * (osl - 1)) / osl`
- Default: `False` (backward compatible)
- All new parameters added at end of function signatures (kwargs-safe)

### Example usage

**CLI (estimate mode):**
\`\`\`bash
aiconfigurator cli estimate \
  --model-path meta-llama/Llama-3.1-8B \
  --system h100_sxm \
  --batch-size 128 \
  --inclusive-tpot
\`\`\`

**CLI (recommend mode):**
\`\`\`bash
aiconfigurator cli recommend \
  --model-path Qwen/Qwen3-32B \
  --system h200_sxm \
  --target-request-rate 50 \
  --inclusive-tpot
\`\`\`

**Python API:**
\`\`\`python
from aiconfigurator.cli.api import cli_recommend

result = cli_recommend(
    model_path="Qwen/Qwen3-32B",
    system="h200_sxm",
    target_request_rate=50.0,
    inclusive_tpot=True,
)
\`\`\`

## Test plan

- [x] All 15 new tests passing
- [x] All existing tests passing (no regression)
- [x] Argument parsing validated for estimate and recommend modes
- [x] Scalar transformation utility tested (formula, edge cases)
- [x] Python API examples added to docstrings
- [x] CLI user guide updated

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added the `--inclusive-tpot` option for estimate, recommend, experiment, and default workflows.
  - Estimate results and EPD logs can display inclusive TPOT based on time to first token, inter-token latency, and output length.
  - API workflows support enabling inclusive TPOT reporting.
  - SLA filtering and internal calculations continue to use inter-token latency.

- **Documentation**
  - Added usage guidance and examples for supported CLI modes and API workflows.
  - Clarified how estimate output reports and transforms TPOT values.

### CODE REVIEW

**Comment by natoscott** [comment]
**Re: EPD mode not applying inclusive TPOT**

Fixed in commit `b8ffd0d46`. The EPD code path now applies the inclusive TPOT transformation before logging output (lines 2599-2602 in `main.py`). 

The transformation is only applied when both `args.inclusive_tpot` is set and the row contains the required `ttft` and `tpot` keys:

```python
# Apply inclusive TPOT transformation if requested
if getattr(args, "inclusive_tpot", False) and "tpot" in row and "ttft" in row:
    from aiconfigurator.cli.report_and_save import get_inclusive_tpot
    row["tpot"] = get_inclusive_tpot(row["ttft"], row["tpot"], args.osl)
```

All EPD tests continue to pass.

**Comment by natoscott** [comment]
**Re: Missing validation for `osl <= 0`**

This is intentional. During adversarial review, we removed all validation from `get_inclusive_tpot()` to match the existing `_apply_inclusive_tpot()` DataFrame approach, which also has no validation.

**Rationale ("fail naturally" philosophy):**
- `EstimateResult` values come from the SDK and should never be invalid in practice
- If they are invalid, that's a bug that should fail loudly (ZeroDivisionError is obvious)
- This simplification removed 44 lines of defensive code and 6 error tests
- Matches existing patterns in the codebase

The DataFrame version (`_apply_inclusive_tpot()`) from the original PR #1141 has the same "no validation" approach - we're being consistent with that design.

If `osl` is ever 0 or negative, that indicates a fundamental problem with the SDK output that should be investigated, not silently handled.

File: src/aiconfigurator/cli/api.py [hunk 1]
```diff
@@ -147,6 +148,7 @@ def _execute_and_wrap_result(
         target_request_rate=target_request_rate,
         target_concurrency=target_concurrency,
         parallel_experiments=parallel_experiments,
+        inclusive_tpot=inclusive_tpot,
```
Comment (jasonqinzhou) [inline src/aiconfigurator/cli/api.py:151]: Please make `inclusive_tpot=True` observable in the returned `CLIResult`, not only in terminal logging and saved CSVs. `_execute_tasks` keeps `best_configs` and `pareto_fronts` raw and uses this flag only inside `log_final_summary`; this wrapper then returns those raw frames unchanged. A focused probe returned TPOT `20.0` with the flag enabled, even though the new examples promise transformed `result.best_configs`. That makes the new option a no-op for a Python caller without `save_dir`. Please transform output copies before returning (while ensuring `save_results` does not apply the formula a second time), or revise the API surface and documentation if raw return values are intentional.

File: src/aiconfigurator/cli/api.py [hunk 2]
```diff
@@ -724,7 +768,9 @@ def cli_exp(
     if not tasks:
         raise ValueError("No valid experiments found in configuration.")
 
-    result = _execute_and_wrap_result(tasks, mode="exp", top_n=top_n)
+    result = _execute_and_wrap_result(
+        tasks, mode="exp", top_n=top_n, parallel_experiments=True, inclusive_tpot=inclusive_tpot
```
Comment (jasonqinzhou) [inline src/aiconfigurator/cli/api.py]: Please remove the new unconditional `parallel_experiments=True` here, or expose it as a separately designed opt-in. Before this PR, `cli_exp()` executed in declaration order, and the CLI `exp` path still does. In parallel mode `_execute_tasks` inserts results as futures complete and uses dictionary order to break equal-throughput ties; a focused two-task probe changed `chosen_exp` from the declared first experiment to the experiment that finished first. This alters existing `cli_exp()` behavior even when `inclusive_tpot=False`. Preserve the previous execution policy for this feature, and handle parallel execution separately with deterministic ordering and coverage if it is desired.

**Review by jasonqinzhou — CHANGES_REQUESTED** [review]
**REQUEST_CHANGES · 5.8/10 · high confidence**

## What this PR does

This PR extends inclusive-TPOT reporting to estimate and recommend CLI modes and exposes the same option through the default, recommend, and experiment Python APIs. It adds a scalar conversion helper, threads the flag through display and save paths, covers EPD estimate output, and documents the new modes.

The output-only separation is thoughtfully preserved for SLA filtering, and the implementation explicitly covers the EPD estimate path while adding focused parsing and formula tests.

## Why this score

The packet is blocked because the mandatory Claude Fable lane could not verify that claude-fable-5 served the request; only fallback Haiku and Sonnet models were observed. Independently, Codex confirmed that returned Python API data remains raw despite the new inclusive-TPOT contract and that cli_exp now forces completion-order-dependent parallel execution. Those candidate findings are preserved below but are not selected for posting from this blocked packet.

**Comment by natoscott** [comment]
Thanks for the careful review — both findings are fair.

**`parallel_experiments=True` in `cli_exp` (api.py):** agreed, that was an accidental regression unrelated to this feature. Removed — `cli_exp()` is back to declaration-order execution.

**Raw returns from `CLIResult`:** you're right that the docstrings were misleading, and I've fixed them. On the direction, some context you may not have had: the original `--inclusive-tpot` change (#1141) deliberately scoped this as *output-only* — its commit message states the transformation is "applied to output copies only (terminal table and saved CSV)" and that "SDK callers receive raw DataFrames and can apply the formula themselves if needed." SLA filtering, pareto computation, and artifact generation all intentionally run on the raw frames. That design was reviewed and approved by a maintainer in #1141.

So rather than diverge and start transforming the returned frames (which would also make these functions inconsistent with `cli_estimate`, which returns a raw `EstimateResult` alongside the `get_inclusive_tpot()` helper), I've kept returns raw and corrected the docs to match that intent: the flag transforms terminal + CSV output, returned DataFrames stay raw, and callers who want the transformed value use `get_inclusive_tpot()`. Added a test locking that contract.

(For the same reason, `get_inclusive_tpot()` intentionally has no `osl > 0` guard — per the #1141 review, osl is never zero in LLM inference, so it "fails naturally" like `_apply_inclusive_tpot`.)

If you'd still prefer the returned frames be transformed, I'm happy to do it — but wanted to flag the original design decision first so we make that change deliberately rather than as a bug fix. Let me know which you'd like.
